frontend/ui/widget/opdsbrowser.lua

```diff
@@ -479,10 +491,7 @@ function OPDSBrowser:downloadFile(item, format, remote_url)
 
     if c == 200 then
         DEBUG("file downloaded to", local_path)
-        UIManager:show(InfoMessage:new{
-            text = _("File saved to:\n") .. local_path,
-            timeout = 3,
-        })
+        self:openBook(local_path)
```

Comment:
It's generally not a good practice to make a widget relying on an app component (readerui in this case) because widgets are the building blocks for apps. So ideally only apps should depend on widgets.

I suggest we add a download success callback attribute to this widget. Then in https://github.com/koreader/koreader/blob/master/frontend/apps/opdscatalog/opdscatalog.lua#L70, we can pass in the book opening logic as a download success callback.

This also makes it possible to close OPDSCatalog before opening the reader. To do so, you can simply create a local reference of the newly created OPDSCatalog before passing it to `UIManager:show`. Then in the download success callback, you call `UIManager:close` on the local reference of OPDSCatalog object before calling into ReaderUI.